Content.Server/AME/AMEFuelContainerSystem.cs

```diff
@@ -0,0 +1,160 @@
+
+        private (int fuel, int capacity) GetAmeFuelContainerFuelAndCapacity(EntityUid uid,
+            AMEFuelContainerComponent? container = null)
+        {
+            return !Resolve(uid, ref container)
+                ? (0, 0)
+                : (container.FuelAmount, container.MaxFuelAmount);
+        }
+
+        private bool TryOpenAMEFuelContainer(EntityUid uid, EntityUid? user,
+            AMEFuelContainerComponent? container = null,
+            ItemComponent? item = null,
+            SinguloFoodComponent? singuloFood = null,
+            PointLightComponent? light = null,
+            AppearanceComponent? appearance = null)
+        {
+            if (!Resolve(uid, ref container, ref singuloFood, ref light, ref appearance))
+                return false;
+
+            if (!container.Sealed)
+                return false;
+
+            Resolve(uid, ref item);
+
+            container.Sealed = false;
+            singuloFood.Energy = SinguloFoodPerThousand;
+            light.Enabled = true;
+
+            appearance.SetData(AMEFuelContainerVisuals.IsOpen, true);
+
+            if (item != null)
+                _itemSystem.SetHeldPrefix(uid, "open", item);
+
+            SoundSystem.Play("/Audio/Items/crowbar.ogg", Filter.Pvs(uid, entityManager: EntityManager), uid);
+
+            if(user != null)
+                _popupSystem.PopupEntity(Loc.GetString("ame-fuel-container-component-on-pry"), uid, Filter.Entities(user.Value));
+
+            container.Dirty();
+
+            _activeAMEFuelContainers.Add(uid);
+            return true;
+        }
+
+        private void OnAMEFuelContainerStartup(EntityUid uid, AMEFuelContainerComponent component,
+            ComponentStartup args)
+        {
+            component.Dirty();
+        }
+
+        private void OnAMEFuelContainerExamine(EntityUid uid, AMEFuelContainerComponent container, ExaminedEvent args)
+        {
+            args.PushMarkup(container.Sealed
+                ? Loc.GetString("ame-fuel-container-component-on-examine-closed-message")
+                : Loc.GetString("ame-fuel-container-component-on-examine-opened-message"));
+
+            if (!args.IsInDetailsRange)
+                return;
+
+            var (fuel, capacity) = GetAmeFuelContainerFuelAndCapacity(uid, container);
+
+            args.PushMarkup(Loc.GetString("ame-fuel-container-component-on-examine-detailed-message",
+                ("colorName", fuel < capacity / FixedPoint2.New(4f) ? "darkorange" : "orange"),
+                ("fuelLeft", fuel),
+                ("fuelCapacity", capacity),
+                ("status", string.Empty)));
+        }
+
+        private void OnAMEFuelContainerInteractUsing(EntityUid uid, AMEFuelContainerComponent container,
+            InteractUsingEvent args)
+        {
+            if (!_toolSystem.HasQuality(args.Used, container.QualityNeeded))
+                return;
+
+            args.Handled = TryOpenAMEFuelContainer(uid, args.User, container);
+        }
```

Comment:
iirc toolsystem has a method for `UseTool` which you should be using instead, also lets you use a doafter if you want. `RepairableSystem` uses this